src/functional/program_processors/codegens/gtfn/gtfn_ir.py

```diff
@@ -155,6 +167,23 @@ class TemporaryAllocation(Node):
     "unstructured_domain",
     "named_range",
 ]
+CPP_FUN_OBJ_BUILTINS = [
+    "std::plus<void>{}",
+    "std::minus<void>{}",
+    "std::multiplies<void>{}",
+    "std::divides<void>{}",
+    "std::equal_to<void>{}",
+    "std::not_equal_to<void>{}",
+    "std::less<void>{}",
+    "std::less_equal<void>{}",
+    "std::greater<void>{}",
+    "std::greater_equal<void>{}",
+    "std::logical_and<void>{}",
+    "std::logical_or<void>{}",
+    "std::bit_xor<void>{}",
+    "std::modulos<void>{}",
+    "std::logical_not<void>{}",
+]
```

Comment:
As we discussed, it might make sense to not use `SymRef`s and create a new node such as `StlFunctor` or something similar. That could reduce modifications and special cases to the existing code. `SymRef`s also don't benefit code generation much, whether a symbol is defined or not is verified by the C++ compiler anyway.